### PR TITLE
ci: Update GHA release workflow name, fixes  issue with publish not pushing to Dockerhub

### PR DESCRIPTION
**Pull request by lorenjohnson**

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -55,6 +55,7 @@ if [ "$PUBLISH" == true ]; then
 	TAGS+=(
 		"${IMAGE_TAGS[@]}"
 	)
+	BUILD_ARGS+=("--push")
```
Comment (rti) [inline build/build.sh:58]: was this broken before because this was missing?

File: .github/workflows/build_publish_image_release.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
           if [[ "$TAG_NAME" =~ ^([^@]+)@([0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?)$ ]]; then
             # Sets and passes the imageName output from this step, which is used in the subsequent step
             # e.g. `steps.extract_tag.outputs.imageName` gets set to "wikibase" matched from "wikibase@1.0.1")
-            echo "::set-output name=imageName::${BASH_REMATCH[1]}"
+            echo "{imageName}=${BASH_REMATCH[1]}" >> $GITHUB_OUTPUT
```
Comment (rti) [inline .github/workflows/build_publish_image_release.yml:32]: ❤️

File: build/build.sh (hunk 1 above)
Comment (rti) [reply build/build.sh:58]: maybe we can mention this fix in the PR title as this ends up in the `main` branch log

**Review by rti — APPROVED** [review]

File: build/build.sh (hunk 1 above)
Comment (lorenjohnson) [reply build/build.sh:58]: Yes, exactly. Tried a test release of an image and things worked as expected, except:

- It only built but didn't push the images, which this fixes
- There was an access error when the script tried to create the GH Release entry which is probably a simple fix, but figured we'd look at that when we get to testing this together

I'll update the title.

File: .github/workflows/build_publish_image_release.yml (hunk 2 above)
Comment (lorenjohnson) [reply .github/workflows/build_publish_image_release.yml:32]: Yeah still not the clearest API, but better.

**Pull request merged**